src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/GRPCConnectionManager.java

```diff
@@ -175,10 +173,12 @@ private int getPortFromHost(final InstanceDetails remoteHost) {
   }
 
   private ManagedChannel buildInsecureChannel(final InstanceDetails remoteHost) {
-    return ManagedChannelBuilder.forAddress(
-            remoteHost.getInstanceIp().toString(),
-            getPortFromHost(remoteHost)
-    ).usePlaintext().build();
+    return ManagedChannelBuilder.forAddress(remoteHost.getInstanceIp().toString(),
+        getPortFromHost(remoteHost))
+                                .usePlaintext()
+                                .enableRetry()
+                                .maxRetryAttempts(MAX_RETRY_ATTEMPTS)
```

Comment:
do we also want to add deadline for rpc call ?

And I also notice that maxRetryAttempts() is marked as experimental. is it going to be an issue to use it inprod ?

Reply:
Deadlines for RPC - We use streaming calls, and deadlines aren't applicable to long lived streaming calls. I don't think there is a simple way to enforce deadlines per message. If we were using a blocking call/unary rpc, then we can add a deadline for it.

Both the enableRetry() and maxRetryAttempts() are marked experimental because of a different reason - its dependency on service config apis. We will consume those when we implement application level retries.